archeryutils/handicaps/handicap_scheme.py

```diff
@@ -216,104 +231,47 @@ def arrow_score(  # noqa: PLR0912 Too many branches
-            s_bar = (
-                9.0
-                - 2.0
-                * sum(
-                    np.exp(-((((n * tar_dia / 10.0) + arw_rad) / sig_r) ** 2))
-                    for n in range(1, 5)
-                )
-                - np.exp(-((((5.0 * tar_dia / 10.0) + arw_rad) / sig_r) ** 2))
-            )
-
-        elif target.scoring_system == "10_zone":
-            s_bar = 10.0 - sum(
-                np.exp(-((((n * tar_dia / 20.0) + arw_rad) / sig_r) ** 2))
-                for n in range(1, 11)
-            )
-
-        elif target.scoring_system == "10_zone_6_ring":
-            s_bar = (
-                10.0
-                - sum(
-                    np.exp(-((((n * tar_dia / 20.0) + arw_rad) / sig_r) ** 2))
-                    for n in range(1, 6)
-                )
-                - 5.0 * np.exp(-((((6.0 * tar_dia / 20.0) + arw_rad) / sig_r) ** 2))
-            )
-
-        elif target.scoring_system == "10_zone_compound":
-            s_bar = (
-                10.0
-                - np.exp(-((((tar_dia / 40.0) + arw_rad) / sig_r) ** 2))
-                - sum(
-                    np.exp(-((((n * tar_dia / 20.0) + arw_rad) / sig_r) ** 2))
-                    for n in range(2, 11)
-                )
-            )
-
-        elif target.scoring_system == "10_zone_5_ring":
-            s_bar = (
-                10.0
-                - sum(
-                    np.exp(-((((n * tar_dia / 20.0) + arw_rad) / sig_r) ** 2))
-                    for n in range(1, 5)
-                )
-                - 6.0 * np.exp(-((((5.0 * tar_dia / 20.0) + arw_rad) / sig_r) ** 2))
-            )
-
-        elif target.scoring_system == "10_zone_5_ring_compound":
-            s_bar = (
-                10.0
-                - np.exp(-((((tar_dia / 40) + arw_rad) / sig_r) ** 2))
-                - sum(
-                    np.exp(-((((n * tar_dia / 20) + arw_rad) / sig_r) ** 2))
-                    for n in range(2, 5)
-                )
-                - 6.0 * np.exp(-((((5 * tar_dia / 20) + arw_rad) / sig_r) ** 2))
-            )
-
-        elif target.scoring_system == "WA_field":
-            s_bar = (
-                6.0
-                - np.exp(-((((tar_dia / 20.0) + arw_rad) / sig_r) ** 2))
-                - sum(
-                    np.exp(-((((n * tar_dia / 10.0) + arw_rad) / sig_r) ** 2))
-                    for n in range(2, 7)
-                )
-            )
-
-        elif target.scoring_system == "IFAA_field":
-            s_bar = (
-                5.0
-                - np.exp(-((((tar_dia / 10.0) + arw_rad) / sig_r) ** 2))
-                - np.exp(-((((3.0 * tar_dia / 10.0) + arw_rad) / sig_r) ** 2))
-                - 3.0 * np.exp(-((((5.0 * tar_dia / 10.0) + arw_rad) / sig_r) ** 2))
-            )
-
-        elif target.scoring_system == "Beiter_hit_miss":
-            s_bar = 1.0 - np.exp(-((((tar_dia / 2.0) + arw_rad) / sig_r) ** 2))
+        return s_bar
```

Comment:
Agreed, think I had that before. Although you were interested before in inlining s_bar directly into the arrow score function, is that still on the cards? Otherwise will implement this

Reply:
Nope, on a proper read and reflection I think keeping separate is better (see review summary).